polaris/polaris/integrations/transactions.py

```diff
@@ -0,0 +1,99 @@
+from typing import Dict, List
+
+from django.db.models import QuerySet
+
+from polaris.models import Transaction
+
+
+class DepositIntegration:
+    """
+    The container class for deposit integration functions.
+
+    Subclasses must be registered with Polaris by passing it to
+    :func:`polaris.integrations.register_integrations`.
+    """
+    @classmethod
+    def poll_pending_deposits(cls, pending_deposits: QuerySet
+                              ) -> List[Transaction]:
+        """
+        **OVERRIDE REQUIRED**
+
+        This function should poll the financial entity for the state of all
+        `pending_deposits` transactions and return the ones ready to be
+        executed on the Stellar network.
+
+        For every transaction that is returned, Polaris will submit it to the
+        Stellar network. If a transaction was completed on the network, the
+        overridable :meth:`after_deposit` function will be called, however
+        overriding this function is optional.
+
+        If the Stellar network is unable to execute a transaction returned
+        from this function, it's status will be marked as ``pending_stellar``
+        and its ``status_message`` attribute will be assigned a description of
+        the problem that occurred.
+
+        `pending_deposits` is a QuerySet of the form
+        ::
+
+            Transactions.object.filter(
+                kind=Transaction.KIND.deposit,
+                status=Transaction.STATUS.pending_user_transfer_start
+            )
+
+        If you have many pending deposits, you may way want to batch
+        the retrieval of these objects to improve query performance and
+        memory usage.
+
+        :param pending_deposits: a django Queryset for pending Transactions
+        """
+        raise NotImplementedError(
+            "`poll_transactions` must be implemented to process deposits"
+        )
+
+    @classmethod
+    def after_deposit(cls, transaction: Transaction):
+        """
+        Use this function to perform any post-processing of `transaction` after
+        its been executed on the Stellar network. This could include actions
+        such as updating other Django models in your project or logging
+        relevant information. Overriding this function is not required
+
+        :param transaction: The django Transaction object that was executed on
```

Comment:
We can if we want to add documentation for the models defined in Polaris. I began doing this originally but realized that most fields were self explanatory, but for columns such as `status`, it would probably be useful.